### ISSUE TITLE
built in sql client keeps asking for password unnecessarily

### ISSUE THREAD

**Issue by tomholub** [cockroachdb/cockroach#26180]
My sql users are all set up to log in exclusively with certificates / keys, no passwords.

When logging in through `cockroach sql`, it will always ask me for a password. No biggie, I just hit enter every time (empty password) to log in.

This becomes an issue when I want to have a script to execute some SQL:

`cockroach sql --user=xxx --database=yyy --execute="SELECT 1;"`

Now it asks for a password which makes scripting difficult. Does not happen to root user, only non-root users.

Again, hitting enter on the password prompt will execute the command.

There should be a `--no-password` option, or better yet, if I'm supplying a cert that can be used for log in, I should not be prompted for a pw.

**Comment by bdarnell** [comment]
Thanks for the report! It looks like we fixed this in #25252 but didn't backport the fix to the release-2.0 branch. We should get that into release 2.0.3 in a couple of weeks.

Unfortunately I can't offer you any workarounds in the meantime except A) using passwords B) using root, or C) wrapping it in something `expect` which can send the enter keypress for you (reading the password fails if the input is not something like a real terminal, so you can't just pipe in `/dev/null` or something like that).

**Comment by tomholub** [comment]
Thanks, sounds good

**Comment by benesch** [comment]
I actually do have a workaround for you: use the `--url` option with a dummy password, like so:

```
./cockroach sql --certs-dir=CERTS-DIR --url=postgres://USER:fakepass@localhost:26257
```

But we'll get the fix into 2.0.3! Sorry for the inconvenience.

**Comment by tomholub** [comment]
@benesch That works for me. Thank you
